docs/proposals/resource-catalog.md

```diff
@@ -0,0 +1,222 @@
+
+
+We can reshape the catalog resource to more easily express them, secure them, and claim them.
+
+# Goals
+ 1.  cluster-admin should control which catalog exist.
+ 2.  cluster-admin should be able/disable per-catalog exposure on a per-project granularity.
+ 3.  project-admin should be able to delegate per-catalog exposure.
+ 4.  project-editor should be able to claim an entry from the catalog.
+ 5.  ResourceCatalogClaims should be sync-able, but not require syncing.
+ 6.  project-admin should be able to control the shape of claim realization and prevent unwanted injection.
+ 7.  ResourceCatalogEntries should not be restricted to specific claim realization mechanism.
+
+To achieve the ACL related goals, we can create namespaced resources that are selected by label at a cluster level.
+
+
+# API Objects
+
+## ResourceCatalog
+ResourceCatalog is a cluster-scoped resource.  It has a cluster-admin controlled spec that is used to drive a controller.
+The spec includes a label selector that is used in a list/watch to filter all ResourceCatalogEntries in the cluster.
+The controller will find matching entries and copy a reference into the ResourceCatalog.Status field.  The controller
```

Comment:
Hrm - going back to the original design that's what we said we'd do.
However, back then it was part of spec so that admins could explicitly
include certain services.

On Mon, Apr 25, 2016 at 11:30 AM, David Eads <notifications@github.com>
wrote:

> In docs/proposals/resource-catalog.md
> <https://github.com/ncdc/kubernetes/pull/8#discussion_r60933740>:
>
> > + 2.  cluster-admin should be able/disable per-catalog exposure on a per-project granularity.
> > + 3.  project-admin should be able to delegate per-catalog exposure.
> > + 4.  project-editor should be able to claim an entry from the catalog.
> > + 5.  ResourceCatalogClaims should be sync-able, but not require syncing.
> > + 6.  project-admin should be able to control the shape of claim realization and prevent unwanted injection.
> > + 7.  ResourceCatalogEntries should not be restricted to specific claim realization mechanism.
> > +
> > +To achieve the ACL related goals, we can create namespaced resources that are selected by label at a cluster level.
> > +
> > +
> > +# API Objects
> > +
> > +## ResourceCatalog
> > +ResourceCatalog is a cluster-scoped resource.  It has a cluster-admin controlled spec that is used to drive a controller.
> > +The spec includes a label selector that is used in a list/watch to filter all ResourceCatalogEntries in the cluster.
> > +The controller will find matching entries and copy a reference into the ResourceCatalog.Status field.  The controller
>
> The status field would probably grow too large at O(1k) - I don't think we
> can rely on ResourceCatalog.Status containing hard references.
>
> I debated making this virtual (its certainly possible), but it complicates
> the resourceversion computation quite a bit and changes there will break
> the etcd watcher cache and any other spot where people assume that
> resourceversion is a monotonically increasing number.
>
> The easiest implementation that removes it introduces the idea of partial
> ordering between resource versions or (more likely) makes arbitrary fixed
> orderings.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/ncdc/kubernetes/pull/8/files/a2acd970c6316ce196c7bb3f24fb57e1bbef647e#r60933740>
>